manifests/eap/ssl.pp

```diff
@@ -29,37 +28,33 @@
     $keystore_alias,
     $keystore_location,
     $keystore_password,
-    $server_certificate_source,
-    $server_certificate_file,
+    $certificates = undef,
 )
 {
     include lightblue::eap::truststore
 
-    # pull certificate from the source
-    file { $server_certificate_file:
-        owner  => 'root',
-        group  => 'root',
-        mode   => '0600',
-        links  => 'follow',
-        source => $server_certificate_source,
-    }
-    #This will create the keystore at the target location, with the alias eap6 to the cert
-    java_ks { "${keystore_alias}:keystore":
-        ensure       => latest,
-        certificate  => $server_certificate_file,
-        private_key  => $server_certificate_file,
-        target       => "${keystore_location}/eap6.keystore",
-        password     => $keystore_password,
-        trustcacerts => true,
-        require      => [ File[$server_certificate_file], Package[$lightblue::eap::package_name] ],
-    }
     file {"${keystore_location}/eap6.keystore":
         owner   => 'jboss',
         group   => 'jboss',
         mode    => '0600',
-        require => Java_ks["${keystore_alias}:keystore"],
+        notify  => Service['jbossas']
+    }
+
+    #### START Puppet iteration black magic
+    # TODO change this to use new Puppet iteration Syntax
+    # when we upgrade to Puppet 4 i.e. $certificates.each
+
+    # Parse certificate hash into an array of hashes
+    $certificate_data = keys($certificates)
```

Comment:
I am not sure of the pros/cons of the approaches, but create_resources is another way you could go I think. I don't know one is better than another, so not a blocker, just a thought/question.